WIP: Truncate remotes

nedsbeds opened the pull request:
Changes proposed:
---------
Add new truncate-history option to deploy command that will remove previous commits from the branch.

Commit:
Add truncate-branch option to deploy command

Commit:
fix a few code standard issues

Commit:
Updating CHANGELOG.md and setting version for 9.2.3.

Commit:
Updating CHANGELOG.md and setting version for 9.2.3.

Commit:
9.2.3 release CHANGELOG cleanup.

Removing extra links from CHANGELOG.md for 9.2.3 release.

grasmash committed:
Update CHANGELOG.md

Commit:
truncate the correct branch when creating a tag deployment

Commit:
Update option name to truncate-history

Commit:
Update option name documentation to truncate-history

Commit:
Explicitly check whether to truncate history in calling functions

sdelbosc commented on src/Robo/Commands/Deploy/DeployCommand.php:
```diff
@@ -653,4 +666,59 @@ public function installDrupal() {
+   */
+  public function truncateBuildRepoHistory($options) {
+
+    if ($options['truncate-history']) {
```
I would test in one place only.
I would probably keep it here.

nedsbeds replied:
I've removed the second check

sdelbosc commented on src/Robo/Commands/Deploy/DeployCommand.php:
```diff
@@ -653,4 +666,59 @@ public function installDrupal() {
+   * Truncates the history from the deployment branch on the remote repo.
+   * This reduces the number of objects in git and drastically reduces
+   * deployment times on much larger projects.
+   * Trigger with --truncate-history option
```
Will that work with truncate-history being configured in yml file of blt?

nedsbeds replied:
I've added a `deploy.truncate_history` config value so this can now be configured in the blt yml file

sdelbosc replied:
Perfect

sdelbosc commented:
Looks promising to me

Commit:
allow configuration of default truncate behaviour with deploy.truncate_history config value

sdelbosc commented:
@nedsbeds if you have tested it already, this is good to go as a pr against blt repo

nedsbeds commented:
> @nedsbeds if you have tested it already, this is good to go as a pr against blt repo

@sdelbosc Thanks,I have tested on a throwaway repo, but would be keen for someone else (most likely Graham) to verify this actually has the intended effect before submitting the PR to BLT.

sdelbosc commented:
Could you share this code with Lubosh as well? I will ask him to test this on Alshaya.

nedsbeds commented:
@sdelbosc I've shared with Lubosh. I think it would be an idea to spin up a clone of the remote repo in a trial AC subscription before running it on the real ACSF repo :)

sdelbosc commented:
If I understand correctly we would impact only one branch at a time. So I am not that scared.

Commit:
add documentation for the --truncate-history feature

mirsoftacquia commented on src/Robo/Commands/Deploy/DeployCommand.php:
```diff
@@ -528,11 +533,17 @@ protected function push($identifier, $options) {
 
+    $arguments = "";
+
+    if ($options['truncate-history']) {
```
@nedsbeds I think this needs `if ($options['truncate-history'] || $this->truncateHistory) {` otherwise it will ignore the blt settings in yml on deploy.

nedsbeds replied:
Ok that should be fixed now :)

Commit:
update push command to force when truncate is set in config

danepowell committed:
Fixes #2932: Inline example of how to require additional settings (#3342)

* Adding a default.includes.settings.php file with instructions for use.

* On setup, copy default.includes.settings.php file to sites/x/settings/ directory.

* PHPCS fix.

* Adding guidance for the use of the includes.settings.php file to onboarding docs.

* Adding a warning line to settings.php to discourage adding settings there.

* Adding default file as example of global settings globbing.

* On setup, copy global.settings.default.php file to sites directory.

* Moving the global.settings.default.php file to the correct location.

* Revising documentation to reflect the global option for settings.

hugovk committed:
9.2.x: Travis CI: Fix for forks (#3333)

Changes proposed:
---------
(What are you proposing we change?)

- Fix Travis CI config so forks can build before submitting PRs to this repo, which currently fail with:

```
$ if [[ "$TRAVIS_PULL_REQUEST" == "false" ]]; then openssl aes-256-cbc -K $encrypted_c0b166e924da_key -iv $encrypted_c0b166e924da_iv -in id_rsa_blt.enc -out ~/.ssh/id_rsa -d; chmod 600 ~/.ssh/id_rsa; ls -lash ~/.ssh; eval "$(ssh-agent -s)"; ssh-add ~/.ssh/id_rsa; fi
iv undefined
```
https://travis-ci.org/hugovk/blt/jobs/479433209

- (To fail fast, and check results before bothering you :)
- Only call the `openssl` command if one of the secret env vars it uses (`$encrypted_c0b166e924da_key`) is set

Steps to replicate the issue:
----------
(If this PR is not fixing a defect/bug, you can remove this section.)

1. Create a fork of this repo
2. Make a build on Travis CI without setting secret encrypted env vars

Steps to verify the solution:
-----------
(How does someone verify that this does what you think does?)

1. Create a fork of this repo
2. Make a build on Travis CI without setting secret encrypted env vars
-> Build passes

Or check my fork: https://travis-ci.org/hugovk/blt/builds/479469643

----

This has been happening for at least 5 months (see below), and this change will allow contributors to contribute more easily and with fewer failing PRs submitted.

![image](https://user-images.githubusercontent.com/1324225/51125207-cf77bc80-1828-11e9-8f3b-5dc910741515.png)

Commit:
Merge branch '9.2.x' into truncate-remotes
